demos/http/http_demo_s3_download_async/http_demo_s3_download_async.c

```diff
@@ -0,0 +1,751 @@
+                                     &pPath,
+                                     &pathLen );
+
+            if( httpStatus != HTTP_SUCCESS )
+            {
+                LogError( ( "An error occurred in getUrlPath() on URL %s. Error code: %d",
+                            S3_PRESIGNED_URL,
+                            httpStatus ) );
+                returnStatus = EXIT_FAILURE;
+            }
+        }
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            /* Retrieve the address location and length from the S3_PRESIGNED_URL. */
+            httpStatus = getUrlAddress( S3_PRESIGNED_URL,
+                                        strlen( S3_PRESIGNED_URL ),
+                                        &pAddress,
+                                        &serverHostLength );
+
+            if( httpStatus != HTTP_SUCCESS )
+            {
+                LogError( ( "An error occurred in getUrlAddress() on URL %s\r\n. Error code %d",
+                            S3_PRESIGNED_URL,
+                            httpStatus ) );
+                returnStatus = EXIT_FAILURE;
+            }
+        }
+
+        /**************************** Connect. ******************************/
+
+        /* Establish TLS connection on top of TCP connection using OpenSSL. */
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            /* Attempt to connect to the HTTP server. If connection fails, retry after
+             * a timeout. Timeout value will be exponentially increased till the maximum
+             * attempts are reached or maximum timeout value is reached. The function
+             * returns EXIT_FAILURE if the TCP connection cannot be established to
+             * broker after configured number of attempts. */
+            returnStatus = connectToServerWithBackoffRetries( connectToServer,
+                                                              &networkContext );
+
+            if( returnStatus == EXIT_FAILURE )
+            {
+                /* Log error to indicate connection failure after all
+                 * reconnect attempts are over. */
+                LogError( ( "Failed to connect to HTTP server %s.",
+                            serverHost ) );
+            }
+        }
+
+        /* Define the transport interface. */
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            transportInterface.recv = Openssl_Recv;
+            transportInterface.send = Openssl_Send;
+            transportInterface.pNetworkContext = &networkContext;
+        }
+
+        /******************** Start queues and HTTP task. *******************/
+
+        /* Start request and response queues */
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            queueSettings.mq_maxmsg = 10;
+            queueSettings.mq_msgsize = sizeof( RequestItem_t );
+
+            requestQueue = mq_open( REQUEST_QUEUE,
+                                    O_CREAT | O_WRONLY,
+                                    0700,
```

Comment:
I think you should make this a macro and define what it means in terms of permissions.